cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsResourceConnector.java

```diff
@@ -662,37 +663,39 @@ public Object collectResourcesToRemove(AuthenticatedContext authenticatedContext
     @Override
     public List<CloudResourceStatus> downscale(AuthenticatedContext auth, CloudStack stack, List<CloudResource> resources, List<CloudInstance> vms,
             Object resourcesToRemove) {
-        List<String> instanceIds = new ArrayList<>();
-        for (CloudInstance vm : vms) {
-            instanceIds.add(vm.getInstanceId());
-        }
-        String asGroupName = cfStackUtil.getAutoscalingGroupName(auth, vms.get(0).getTemplate().getGroupName(),
-                auth.getCloudContext().getLocation().getRegion().value());
-        DetachInstancesRequest detachInstancesRequest = new DetachInstancesRequest().withAutoScalingGroupName(asGroupName).withInstanceIds(instanceIds)
-                .withShouldDecrementDesiredCapacity(true);
-        AmazonAutoScalingClient amazonASClient = awsClient.createAutoScalingClient(new AwsCredentialView(auth.getCloudCredential()),
-                auth.getCloudContext().getLocation().getRegion().value());
-        try {
-            amazonASClient.detachInstances(detachInstancesRequest);
-        } catch (AmazonServiceException e) {
-            if (!"ValidationError".equals(e.getErrorCode())
-                    || !e.getErrorMessage().contains("not part of Auto Scaling")
-                    || instanceIds.stream().anyMatch(id -> !e.getErrorMessage().contains(id))) {
-                throw e;
+        if (!vms.isEmpty()) {
+            List<String> instanceIds = new ArrayList<>();
+            for (CloudInstance vm : vms) {
+                instanceIds.add(vm.getInstanceId());
             }
-            LOGGER.info(e.getErrorMessage());
-        }
-        AmazonEC2Client amazonEC2Client = awsClient.createAccess(new AwsCredentialView(auth.getCloudCredential()),
-                auth.getCloudContext().getLocation().getRegion().value());
-        try {
-            amazonEC2Client.terminateInstances(new TerminateInstancesRequest().withInstanceIds(instanceIds));
-        } catch (AmazonServiceException e) {
-            if (!"InvalidInstanceID.NotFound".equals(e.getErrorCode())) {
-                throw e;
+            String asGroupName = cfStackUtil.getAutoscalingGroupName(auth, vms.get(0).getTemplate().getGroupName(),
```

Comment:
The detach and the terminate instances could be individual methods.

Reply:
fixed